Marlin/src/feature/pause.cpp

```diff
@@ -252,94 +256,182 @@ void wait_for_filament_reload(const int8_t max_beep_count/*=0*/) {
-      while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_WAIT_FOR) idle(true);
-      KEEPALIVE_STATE(IN_HANDLER);
-
-      extrude_length = ADVANCED_PAUSE_EXTRUDE_LENGTH;
-
-      // Keep looping if "Extrude More" was selected
-    } while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE);
-
+    if (swap_spool_disabled){ 
+        
+	  do {
+	    if (extrude_length > 0) {
+		  // "Wait for filament extrude"
+		  if (swap_spool_disabled) lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_EXTRUDE); 
+		  // Extrude filament to get into hotend
+          do_pause_e_move(extrude_length, PAUSE_EXTRUDE_F);
+	    }
+		// Show "Extrude More" / "Resume" menu and wait for reply
+		KEEPALIVE_STATE(PAUSED_FOR_USER);
+		wait_for_user = false;
+		lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_OPTION);
+	    while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_WAIT_FOR) idle(true);
+		KEEPALIVE_STATE(IN_HANDLER);
+		
+		extrude_length = PAUSE_EXTRUDE;
+		// Keep looping if "Extrude More" was selected
+	  } while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE);
+  }
+  else do_pause_e_move(extrude_length, PAUSE_EXTRUDE_F);
   #endif
-
+  
+  // Extruder purged = initialisation  
+  planner.set_e_position_mm(0); current_position[E_AXIS] = 0; destination[E_AXIS]=0;
+  
   #if ENABLED(ULTIPANEL)
     // "Wait for print to resume"
-    lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_RESUME);
+    if (swap_spool_disabled)lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_RESUME);
   #endif
-
-  // Set extruder to saved position
-  planner.set_e_position_mm((current_position[E_AXIS] = resume_position[E_AXIS]));
-
+	
+  // Stop blowing after delay
+  #ifdef PAUSE_DWELL
+    delay(PAUSE_DWELL);
+  #endif
+  #if (PAUSE_FANSPEED >0 && PAUSE_FAN < FAN_COUNT)
+    fanSpeeds[PAUSE_FAN]=0 ;
+  #endif  
+  
+  #ifdef PAUSE_RET
+  // Intelligent Retractation before return to print
+  // If resume position negative then travel back Retraction = resume_position
+  // Rare but possible : can go on pause with retracted/swapped status or negative position
+  #ifdef PAUSE_RET
+    #if ENABLED(FWRETRACT)
+	  //If retracted/swapped goto negative position
+	  if (fwretract.retracted_swap[active_extruder]) {
+	    const float tmp_length=fwretract.swap_retract_length;
+	    const float tmp_feed=fwretract.retract_feedrate_mm_s;
+	    do_pause_e_move(tmp_length, tmp_feed); 
+	  }
+	  // If retracted or not : normal fwretract used
+	  else {
+	    const float tmp_length=fwretract.retract_length;  
+            const float tmp_feed=fwretract.retract_feedrate_mm_s;   
+            do_pause_e_move(tmp_length, tmp_feed); 
+	  }
+    #else
+	  //If negative and more than PAUSE_RET goto resume_position 
+	  if (resume_position[E_AXIS]<(-PAUSE_RET)) do_pause_e_move(resume_position[E_AXIS], PAUSE_RET_F);
+          else do_pause_e_move(-PAUSE_RET, PAUSE_RET_F);
+    #endif
+  #endif	
+ 
```

Comment:
The intelligent process is simple
If Epos was retracted or negative before go on pause , we use the same value for the travel back retractation
If EPOS was zero or positive ,  i retract with the pause_ret